src/include/optimizer/abstract_optimizer.h

```diff
@@ -43,6 +44,8 @@ class AbstractOptimizer {
       const std::unique_ptr<parser::SQLStatementList> &parse_tree) = 0;
 
   virtual void Reset(){};
+
+  concurrency::Transaction *txn = nullptr;
```

Comment:
1. Move this to private
2. Initialize in constructor
3. Remove "transaction.h" include and forward declare `concurrency::Transaction`